### PR TITLE
Updates conda versioning to install correct dependencies, changes CI script to better track deps from individual build installs

### PR DESCRIPTION
**Pull request by seunghwak**
* Updates versioning in conda recipes so the correct dependencies are installed, specifically UCX-Py and CTK.  These were causing CI to fall back to installs from the nightly channel instead of the packages built from the PR sources, since the older nightlies were the only ones compatible. This also resulted in the wrong CTK version to be installed in some cases.
* Changes CI script to install each built package individually so log output shows the deps installed for each one. If all packages are installed in one install line, developers debugging CI can't tell which packages caused certain dependencies to be installed.

### CODE REVIEW

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by ajschmidt8** [comment]
@seunghwak, the message below also applies to `DO NOT MERGE` PRs.

---

Please use GitHub's _Draft PR_ feature instead of `WIP` tags in the future. Draft PRs have the benefit of preventing notifications to `codeowners` until PRs are marked _Ready for Review_, which helps cut down on excessive notifications for PRs that are still being worked on. CI will still run on Draft PRs.

Some useful information about Draft PRs:

- https://github.blog/2019-02-14-introducing-draft-pull-requests/
- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests
- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request

**Review by raydouglass — APPROVED** [review]

**Comment by rlratzel** [comment]
@gpucibot merge

**Pull request merged**
